server/bleep/src/indexes/file.rs

```diff
@@ -111,6 +119,7 @@ impl Indexable for File {
                     repo_disk_path: &repo.disk_path,
                     repo_ref: reporef.to_string(),
                     repo_name: &repo_name,
+                    file_filter: &file_filter,
```

Comment:
Do we need to pass the `file_filter` here, or can we just call this `indexed: bool` and bring this calculation:
```
let indexed = file_filter
            .is_allowed(relative_path)
            .unwrap_or_else(|| should_index(relative_path));
```           
 up into this method?

Reply:
We actually use this for cache key calculations as well, otherwise the forced reindexing logic won't work. It is better to keep this for this PR.

Reply:
Could we not have an `indexed: bool` field on `Workload` and calculate whether a file should be indexed in `index_repository` (before instantiating a `Workload`)?

Reply:
looking at this again, we don't only need a binary choice, since we ideally don't want to reindex everything with a new index key. so we need a ternary choice: denied, included, or not mentioned in the ruleset at all.